Neos.Neos/Classes/Controller/Service/NodesController.php

```diff
@@ -107,9 +107,21 @@ public function indexAction($searchTerm = '', array $nodeIdentifiers = array(),
             }
         }
 
+
+
         $contentContext = $this->createContentContext($workspaceName, $dimensions);
+        $nodes = [];
+
+        //If there is a Node with $searchTerm as identifier, also return it.
+        if ($contentContext->getNodeByIdentifier($searchTerm) instanceof NodeInterface) {
```

Comment:
would make sense to check if it's a valid identifier first before looking up in the database like
``if (preg_match(NodeIdentifierValidator::PATTERN_MATCH_NODE_IDENTIFIER, $value) !== 0) {``

Reply:
Makes sense! Added it...